tests/test_e2e.py

```diff
@@ -56,12 +56,29 @@ def _infer(self, model, prompt=None, protocol=None):
             args += ["-i", protocol]
         run(args)
 
-    def _profile(self, model, protocol=None, backend=None):
+    def _profile(self, model, protocol=None, backend=None, is_llm=False):
         args = ["profile", "-m", model]
-        if protocol:
-            args += ["-i", protocol]
         if backend:
             args += ["--backend", backend]
+        if is_llm:
+            args += ["--task", "llm"]
+        else:
+            if protocol:
+                args += ["-i", protocol]
+        run(args)
+
+    def _optimize(self, model):
```

Comment:
I think this is failing the `pytest` runs in the github-side checks because there is no `model-analyzer` command found. Can you add the installation as a dependency in pyproject.toml and any other test fix necessary to see if we can get the :heavy_check_mark: on github side?

Reply:
ex: https://github.com/triton-inference-server/triton_cli/actions/runs/8914215110/job/24481334037?pr=52

```
test_e2e.py:82: in _optimize
...
args = ['model-analyzer', 'profile', '--profile-models', 'add_sub', '--triton-launch-mode', 'local', ...]
...
E               FileNotFoundError: [Errno 2] No such file or directory: 'model-analyzer'
```

Reply:
Thanks for the suggestion, already done. :) I added it on the unit testing and documentation [PR (draft)](https://github.com/triton-inference-server/triton_cli/pull/53) that I plan to merge into this one once ready and approved. But I can move that work here so that this is passing even without those PRs.

I assume I'll need to clone and install GenAI-Perf more manually, since it doesn't have a pip wheel. Apologies that the these changes are not in place yet. I'm learning a lot about the triton_cli and the tool packaging this week. Originally, I thought this would be a much harder blocker than it is (i.e. that these tests might run in a Triton container or that I would need to get perf_analyzer installed on the GitHub action side). I can work through this.